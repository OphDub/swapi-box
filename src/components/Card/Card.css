.card {
  background: rgba(0, 0, 0, 0.8);
  border: solid 1px #FFFFFF;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 300px;
  min-height: 375px;
  max-height: 375px;
  margin: 10px 0; }

.card-info {
  list-style-type: none;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 50%;
  padding: 15px; }

.card-li {
  font-weight: 700; }

.favorited {
  background-color: rgba(65, 185, 225, 0.5); }

.favorites,
.people,
.planets,
.vehicles {
  padding: 15px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%; }

.favorites {
  background-color: rgba(153, 99, 153, 0.5); }

.people {
  background: rgba(162, 109, 162, 0.5); }

.planets {
  background: rgba(5, 74, 179, 0.5); }

.vehicles {
  background: rgba(0, 92, 18, 0.5); }
