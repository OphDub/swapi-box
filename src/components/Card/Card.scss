.card {
  border: yellow 1px solid;
}