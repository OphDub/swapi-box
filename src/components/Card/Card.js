import React from 'react';
import './Card.css';
import { object, func } from 'prop-types';

const Card = ({ type, element, saveFavorite }) => {
  const keys = Object.keys(element).filter((key)=>
    key !== 'name' && key !== 'favorited'
  );
  const cardClass = element.favorited ? 'favorited' : ''

  const list = keys.map((key, index) => {
    let info = element[key];
    if (typeof element[key] === 'object') {
      info = element[key].join(', ')
    }

    return (
      <li key={`${Date.now()}${index}`}>
        {key}: {info}
      </li>
    );
  });

  return (
    <article className="card">
      <div className={type}>
        <h3>{element.name}</h3>
        <button onClick={() => saveFavorite(element)}
                className={cardClass}>
          Favorite
        </button>
      </div>
      <ul>
        {list}
      </ul>
    </article>
  );
};

Card.propTypes = {
  element: object.isRequired,
  saveFavorite: func.isRequired,
};

export default Card;