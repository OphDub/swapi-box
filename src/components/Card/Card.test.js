/* eslint-disable */
import React from 'react';
import Card from './Card';

describe('CARD', () => {
  it('should match the snapshot', () => {

  });

  it('should be able to list all the key value pairs of an object', () => {

  })

  it('should call saveFavorite on button click', () => {

  });
});