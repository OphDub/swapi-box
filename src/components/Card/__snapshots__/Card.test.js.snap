// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    element={
        Object {
            "class": "wheeled",
            "favorited": false,
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
          }
    }
    saveFavorite={[Function]}
    type="vehicle"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="vehicle"
>
          <h3>
                    Sand Crawler
          </h3>
          <button
                    className=""
                    onClick={[Function]}
          >
                    Favorite
          </button>
</div>,
        <ul>
          <li>
                    model
                    : 
                    Digger Crawler
          </li>
          <li>
                    passengers
                    : 
                    30
          </li>
          <li>
                    class
                    : 
                    wheeled
          </li>
</ul>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Sand Crawler
</h3>,
            <button
              className=""
              onClick={[Function]}
>
              Favorite
</button>,
          ],
          "className": "vehicle",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sand Crawler",
            },
            "ref": null,
            "rendered": "Sand Crawler",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              model
              : 
              Digger Crawler
</li>,
            <li>
              passengers
              : 
              30
</li>,
            <li>
              class
              : 
              wheeled
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "15178606924930",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "model",
                ": ",
                "Digger Crawler",
              ],
            },
            "ref": null,
            "rendered": Array [
              "model",
              ": ",
              "Digger Crawler",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "15178606924931",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "passengers",
                ": ",
                "30",
              ],
            },
            "ref": null,
            "rendered": Array [
              "passengers",
              ": ",
              "30",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "15178606924932",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "class",
                ": ",
                "wheeled",
              ],
            },
            "ref": null,
            "rendered": Array [
              "class",
              ": ",
              "wheeled",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="vehicle"
>
            <h3>
                        Sand Crawler
            </h3>
            <button
                        className=""
                        onClick={[Function]}
            >
                        Favorite
            </button>
</div>,
          <ul>
            <li>
                        model
                        : 
                        Digger Crawler
            </li>
            <li>
                        passengers
                        : 
                        30
            </li>
            <li>
                        class
                        : 
                        wheeled
            </li>
</ul>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Sand Crawler
</h3>,
              <button
                className=""
                onClick={[Function]}
>
                Favorite
</button>,
            ],
            "className": "vehicle",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sand Crawler",
              },
              "ref": null,
              "rendered": "Sand Crawler",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                model
                : 
                Digger Crawler
</li>,
              <li>
                passengers
                : 
                30
</li>,
              <li>
                class
                : 
                wheeled
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "15178606924930",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "model",
                  ": ",
                  "Digger Crawler",
                ],
              },
              "ref": null,
              "rendered": Array [
                "model",
                ": ",
                "Digger Crawler",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "15178606924931",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "passengers",
                  ": ",
                  "30",
                ],
              },
              "ref": null,
              "rendered": Array [
                "passengers",
                ": ",
                "30",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "15178606924932",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "class",
                  ": ",
                  "wheeled",
                ],
              },
              "ref": null,
              "rendered": Array [
                "class",
                ": ",
                "wheeled",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
